include/openthread.h

```diff
@@ -868,8 +868,21 @@ typedef void (*otStateChangedCallback)(uint32_t aFlags, void *aContext);
  * @param[in]  aCallback  A pointer to a function that is called with certain configuration or state changes.
  * @param[in]  aContext   A pointer to application-specific context.
  *
+ * @retval kThreadError_None    Added the callback to the list of callbacks.
+ * @retval kThreadError_NoBufs  Could not add the callback due to resource constraints.
+ *
+ */
+ThreadError otSetStateChangedCallback(otInstance *aInstance, otStateChangedCallback aCallback, void *aContext);
```

Comment:
I mean, you would index the callback based on just the `aContext` passed in. A `aContext` passed in with a valid `aCallback` would be an add/update to a callback registration; while a `aContext` with a NULL `aCallback` would be a removal for that `aContext` callback registration.